hbase_storage/src/main/java/com/splicemachine/access/hbase/HBasePartitionAdmin.java

```diff
@@ -720,4 +724,65 @@ private UserPermission getPermission(List<UserPermission> permissions, String us
         }
         return null;
     }
+
+    static public int getPriorityShouldHave(HTableDescriptor td)
+    {
+        String s = td.getValue("tableDisplayName");
+        if( s == null ) {
+            return HConstants.HIGH_QOS;
+        }
+        else {
+            if (s.startsWith("SYS") || s.startsWith("splice:") || s.equals("MON_GET_CONNECTION")) {
+                return HConstants.ADMIN_QOS;
+            } else {
+                return HConstants.NORMAL_QOS;
+            }
+        }
+    }
+
+    @Override
+    public int upgradeTablePriorities() throws Exception {
+        return upgradeTablePriorities(admin);
+    }
+
+    public static int upgradeTablePriorities(Admin admin) throws Exception
+    {
+        final int NUM_THREADS = 10;
+        ExecutorService executor = null;
+        try {
+            HTableDescriptor[] tableDescriptors = admin.listTables();
+
+            List<Callable<Void>> upgradeTasks = Arrays.stream(tableDescriptors)
+                    .filter( td -> td.getPriority() != getPriorityShouldHave(td))
+                    .map( td -> (Callable<Void>) () -> {
+                        setHTablePriority(admin, td, getPriorityShouldHave(td));
+                        return null;
+                    })
+                    .collect(Collectors.toList());
+            executor = Executors.newFixedThreadPool( NUM_THREADS );
```

Comment:
We only need to set priority for splice system tables, so no need to check for all tables in HBase, especially when there are 100K+ tables on the cluster. It will take a while to read all the table descriptors.

You can get all tables and indexes in schema SYS, SYSIBM, and SYSIBMADMIN are available at DataDicitonaryImpl.coreInfo and DataDicitonaryImpl.noncoreInfo. Other important tables are HBaseBackupUtils.spliceSystemTables. Only processing these tables can make upgrade time independent of cluster size. An example to add a property value to table descriptor is UpgradeScriptToAddCatalogVersion.

Reply:
ok, fixed. sdd.upgradeTablePriorities will have a list of tables to check